src/v/raft/state_machine_manager.cc

```diff
@@ -444,6 +460,40 @@ void state_machine_manager::maybe_start_background_apply(
 ss::future<> state_machine_manager::background_apply_fiber(
   entry_ptr entry, ssx::semaphore_units units) {
     while (!_as.abort_requested() && entry->stm->next() < _next) {
+        if (entry->stm->next() < _raft->start_offset()) {
+            auto snapshot = co_await _raft->open_snapshot();
+            if (!snapshot) {
+                vlog(
+                  _log.warn,
```

Comment:
is there a circumstance where we expect this to happen? I am wondering why warn over error. I guess it will retry every 100ms, which should be fine?

Reply:
Only when partition is deleted